linera-core/src/unit_tests/test_utils.rs

```diff
@@ -503,6 +513,27 @@ where
 
         sender.send(certificate_hash)
     }
+
+    async fn do_blobs_last_used_by(
+        self,
+        blob_ids: Vec<BlobId>,
+        sender: oneshot::Sender<Result<Vec<CryptoHash>, NodeError>>,
+    ) -> Result<(), Result<Vec<CryptoHash>, NodeError>> {
+        let validator = self.client.lock().await;
+        let blob_states = validator
+            .state
+            .storage_client()
+            .read_blob_states(&blob_ids)
+            .await;
+        let hashes = match blob_states {
+            Err(err) => Err(err.into()),
+            Ok(blob_states) => Ok(blob_states
+                .into_iter()
+                .map(|blob_state| blob_state.last_used_by)
+                .collect()),
+        };
```

Comment:
This seems cleaner/more idiomatic:

```suggestion
        let hashes = blob_states
            .map(|states| states.into_iter().map(|state| state.last_used_by).collect())
            .map_err(Into::into);
```

Reply:
Correct. Done.